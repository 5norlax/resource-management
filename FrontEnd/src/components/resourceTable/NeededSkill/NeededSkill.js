import React from 'react';
import { Row, Badge, Button } from 'reactstrap';

export default function NeededSkill({ skills, removeSkill }) {

    const style = {
        minHeight: '15px',
        marginRight: '5px',
        marginLeft: '5px',
        marginTop: '5px',
        marginBottom: '5px'
    }
    
    const badges = (skills) => {
        let badges = []
        if (skills.length === 0) {}
        else badges = skills.map((skill) => (
            <Button style={style} key={skill} className="shadow-none" color="secondary" outline>
                <Badge onClick={() => { removeSkill(skill) }} color="secondary">X</Badge> {skill}
            </Button>
        ))
        return badges
    }
    return (
        <Row className="col-md-12">
            {badges(skills)}
        </Row>
    )
};