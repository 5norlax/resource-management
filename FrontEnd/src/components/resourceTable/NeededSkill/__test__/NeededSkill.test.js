import React from 'react'
import ReactDOM from 'react-dom'
import { render, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import NeededSkill from '../NeededSkill'

afterEach(cleanup)

describe('Test NeededSkill', () => {

  it('matches snapshot', () => {
    const tree = renderer.create(<NeededSkill skills={[ 'Java' ]} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<NeededSkill skills={[ 'Java' ]} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

})