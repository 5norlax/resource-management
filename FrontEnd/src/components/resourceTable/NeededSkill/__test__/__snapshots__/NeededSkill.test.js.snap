// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NeededSkill matches snapshot 1`] = `
<div
  className="col-md-12 row"
>
  <button
    aclassName="shadow-none"
    aria-label={null}
    className="btn btn-outline-secondary"
    onClick={[Function]}
    style={
      Object {
        "marginBottom": "5px",
        "marginLeft": "5px",
        "marginRight": "5px",
        "marginTop": "5px",
        "minHeight": "15px",
      }
    }
  >
    <span
      className="badge badge-secondary"
      onClick={[Function]}
    >
      X
    </span>
     
    Java
  </button>
</div>
`;
